### PR TITLE
feat: remove Search interface requirement

### PR DESCRIPTION
**Pull request by gabrielfs7**
Register search using proxy

https://oat-sa.atlassian.net/browse/ADF-258

- [ ] https://github.com/oat-sa/tao-core/pull/2933
- [ ] https://github.com/oat-sa/extension-tao-publishing/pull/111
- [ ] https://github.com/oat-sa/extension-tao-advanced-search/pull/27

### CODE REVIEW

File: src/Action/InitElasticSearch.php [hunk 1]
```diff
@@ -125,10 +122,21 @@ public function __invoke($params): Report
...
+
+            /** @var SearchProxy $search */
+            $search = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
+            
+            if ($search instanceof GenerisSearch) {
+                $this->getServiceManager()->register(Search::SERVICE_ID, $elasticSearch);
+            }
+            
+            if ($search instanceof SearchProxy) {
+                $search->setOption(SearchProxy::OPTION_ADVANCED_SEARCH_CLASS, $elasticSearch);
```
Comment (boajer) [inline src/Action/InitElasticSearch.php:136]: Just a note to update tao-core version on composer before merge

 - [ ] Update tao-core

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on http://aadvanced-search-integration.playground.kitchen.it.taocloud.org:48321/

**Pull request merged**